apiserver/application/application.go

```diff
@@ -339,6 +340,73 @@ func (api *API) Update(args params.ApplicationUpdate) error {
 	return nil
 }
 
+// UpdateApplicationSeries updates the application series.  Series for
+// subordinates updated too.
+func (api *API) UpdateApplicationSeries(args params.ApplicationUpdateSeries) error {
+	if err := api.checkCanWrite(); err != nil {
+		return err
+	}
+	app, err := api.backend.Application(args.ApplicationName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if !app.IsPrincipal() {
+		return errors.New(fmt.Sprintf("%s is a subordinate application, series update is not supported.", args.ApplicationName))
+	}
+
+	units, err := app.AllUnits()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// all units will have the same subordinates, therefore we only
+	// need to check one.
+	unit := units[0]
+	if unit != nil {
+		// Find our subordinate applications, to verify their series as well,
+		for _, subUnitName := range unit.SubordinateNames() {
+			subordinateApplicationName := strings.Split(subUnitName, "/")[0]
+			subordinateApplication, err := api.backend.Application(subordinateApplicationName)
+
+			if err != nil {
+				return errors.Trace(err)
+			}
+			subordinateCharm, _, err := subordinateApplication.Charm()
+			if err != nil {
+				return errors.Trace(err)
+			}
+			_, seriesSupportedErr := charm.SeriesForCharm(args.Series, subordinateCharm.Meta().Series)
+			if seriesSupportedErr != nil && !args.Force {
+				return errors.New(fmt.Sprintf("series %s is not supported by the subordinates of the application in use. Use --force to override.", args.Series))
```

Comment:
Errorf.
Also, the apiserver layer is not the place to put error messages. You need to define a new error type and have the caller react to that. The API is called by more than just the CLI. eg Juju Python library. There's no such thing as --force there.
See apiserver/params/apierror.go